File: 1 Algorithmic Toolbox/Homeworks/2 Algorithmic Warm-up/6 Least Common Multiple/C++/lcm.cpp
#include <iostream>

unsigned long long int gcd(unsigned long long int a, unsigned long long int b) {
	if (b == 0) return a;
	unsigned long long int a_remainder = a % b;
	return gcd(b, a_remainder);
}

unsigned long long int lcm(unsigned long long int a, unsigned long long int b) {
	return a == b && b == 0 ? a : a * b / gcd(a, b);
}

int main() {
	unsigned long long int a(0), b(0);
	std::cin >> a >> b;
	std::cout << lcm(a, b) << std::endl;
	return 0;
}

File: 1 Algorithmic Toolbox/Homeworks/2 Algorithmic Warm-up/4 Least Common Multiple/lcm.py
# Uses python3


def gcd(a, b):
    if b == 0:
        return a
    a_remainder = a % b
    return gcd(b, a_remainder)


def lcm(a, b):
    if a == b == 0:
        return 0
    return a * b // gcd(a, b)


if __name__ == '__main__':
    a, b = map(int, input().split())
    print(lcm(a, b))
